app/src/main/java/com/gruppo4/SMSApp/MainActivity.java

```diff
@@ -0,0 +1,140 @@
+package com.gruppo4.SMSApp;
+
+/**
+ * @author Francesco Bau'
+ */
+
+import com.gruppo4.audioUtility.AudioUtilityManager;
+
+import android.content.Context;
+import android.media.AudioManager;
+import android.os.Bundle;
+import android.os.Vibrator;
+import android.view.View;
+import android.widget.Toast;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+
+public class MainActivity extends AppCompatActivity {
+
+
+    // Default app volume, expressed in percentage
+    private static final int DEFAULT_APP_VOLUME = 50;
+    // The selected Stream.
+    private static AudioUtilityManager.AUMStream stream;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        //Default choice: Stream ALARM
+        stream = AudioUtilityManager.AUMStream.ALARM;
+
+        // Action performed by clicking button0
+        // -> TEST the method getVolume(Context,String).
+        // -> TEST the method setVolume(Context,String,int).
+        findViewById(R.id.button0).setOnClickListener(new View.OnClickListener() {
```

Comment:
Button 0 to 5 have no name to identify them, so I'd suggest either giving them a unique id in the xml layout or assigning the result of findViewById to a named variable. The first option would be more consistent with the xml-named buttons.

Reply:
Thank you for pointing this out. You are right. But button0, button1, ... were made like this on purpose, since they are just like "Test0", "Test1", "Test2", ... , and these methods had plenty of changes, back in these days... And, as you can see, Radio Buttons got their defined IDs, since these Radio Buttons have only one precise scope, which is to select one single Stream. For the regular buttons, if you could see the history of the changes made for these methods , you'd know what am I talking about. These methods changed a lot...
Then again, thank you for pointing this out.
Even though my MainActivity would eventually be destroyed, because what my team needs is just the Utility class, I needed some constructive criticism about my layout, since I'm not really good at it...

Reply:
I totally agree with the @RicDeZen comment.